packages/fuels-accounts/src/provider/retryable_client.rs

```diff
@@ -304,5 +304,11 @@ impl RetryableClient {
         })
         .await
     }
+
+    pub async fn contract_exists(&self, contract_id: &ContractId) -> RequestResult<bool> {
+        self.wrap(|| self.client.contract(contract_id))
+            .await
+            .map(|contract| contract.is_some())
```

Comment:
On my phone, but can you check that this doesn't download the whole code just to check if a contract exists.
If so, maybe let ahmedm s pr go in first and then add a custom query that will minimize the fetched data?

Reply:
I was thinking about this exactly: this query definitely downloads the whole `bytecode: HexString`. Thanks for the tip, I'll wait on #1529.

Reply:
Added a custom query in 196d30f.